repository/src/main/java/org/apache/atlas/repository/store/graph/v2/UniqueQNAttributeMigrationService.java

```diff
@@ -0,0 +1,84 @@
+package org.apache.atlas.repository.store.graph.v2;
+
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.repository.graph.GraphHelper;
+import org.apache.atlas.repository.graphdb.AtlasVertex;
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class UniqueQNAttributeMigrationService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UniqueQNAttributeMigrationService.class);
+
+    private final EntityGraphRetriever entityRetriever;
+
+
+    private Set<String> entityGuids;
+    private final TransactionInterceptHelper   transactionInterceptHelper;
+
+    private final String QUALIFIED_NAME_ATTR = "qualifiedName";
+    private final String UNIQUE_QUALIFIED_NAME_ATTR = "__u_qualifiedName";
+
+    public UniqueQNAttributeMigrationService(EntityGraphRetriever entityRetriever, Set<String> entityGuids, TransactionInterceptHelper transactionInterceptHelper) {
+        this.entityRetriever = entityRetriever;
+        this.transactionInterceptHelper = transactionInterceptHelper;
+        this.entityGuids = entityGuids;
+    }
+
+    public void migrateQN() throws Exception {
+        try {
+            for (String entityGuid : entityGuids) {
+                AtlasVertex entityVertex = entityRetriever.getEntityVertex(entityGuid);
+
+                if (entityVertex == null) {
+                    LOG.error("Entity vertex not found for guid: {}", entityGuid);
+                    continue;
+                }
+
+                boolean isCommitRequired = migrateuniqueQnAttr(entityVertex);
+                if (isCommitRequired){
+                    LOG.info("Committing changes for entity: {}", entityGuid);
+                    commitChanges();
```

Comment:
Seems like we are committing each vertex, if this is true, please change the logic to do it in bulk